osmaxx/excerptexport/models/export.py

```diff
@@ -80,11 +80,17 @@ def status_update_url(self):
 
     def set_and_handle_new_status(self, new_status, *, incoming_request):
         assert new_status in dict(self.STATUS_CHOICES)
+        if self.status == new_status and self.last_update_overdue():
+            new_status = self.FAILED
```

Comment:
IMHO I think this should always hold true, even if `osmaxx.job_progress.views.tracker()` might be a very unlikely place that a status arrives that is the same as the already existing status.